### PR TITLE
added shortcodes for direct use in posts

### PR DESCRIPTION
**Pull request by jonocarroll**
`shortcode`s seem like an easy addition here, allowing direct use in individual posts via [get_the_github_view_link]. Seems cleaner than adding it to the default theme for all posts if that's not what you want. Tested on my own WP site and works. Official tests not performed.

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
Thanks! This is a great addition, although I think we should go with something a little shorter, rather than mirroring the function name. What about just `[github_view_link]` etc?

**Comment by jonocarroll** [comment]
Absolutely. The `shortcode` should be short (I figured I'd start with the clearest transition). What about just one `shortcode` with an attribute or several? I did one for each function but you could combine them;

    [github type="link" text="As seen on GitHub"]

with the function chosen via combination of attributes. For the simpler implementation, yes, dropping "get_the" from the `shortcode` would be good.

**Comment by mAAdhaTTah** [comment]
I dig it. Could you also include documentation for the new shortcode? Thanks for the contribution.

**Comment by jonocarroll** [comment]
Can do. FYI; https://codex.wordpress.org/Shortcode_API .

Also, I'm not sure why Travis CI is failing. Something to do with composer and autoload, but I'm not familiar enough with your build to diagnose.

**Comment by jonocarroll** [comment]
I've added the new multi-parameter function and corresponding shortcode `[wpghs type='link' text='abc']`, with some documentation added to the README. I'll need someone to investigate any Travis CI issues.

Tested all combinations of options on my own Wordpress site and all seems good.

**Comment by mAAdhaTTah** [comment]
Yeah, I think the Travis bugs are related to the the PHP 5.2-compatible autoloader. I'll have to investigate that separately.

**Comment by mAAdhaTTah** [comment]
I'll probably review, merge, and publish a new version of the plugin this weekend. Thanks again!

**Comment by jonocarroll** [comment]
Okay, last changes made, good to go for review/merge.

I've added a `target` attribute back in to allow linking to either the view or edit URL. Options tested on my own Wordpress site and seem to do what I expect. Documentation updated again.

**Comment by mAAdhaTTah** [comment]
Thanks.

One quick note: We follow the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/). Which first means tabs > spaces. You can check the Scrutinizer details in this PR to see what other changes may be needed on that front.

**Comment by jonocarroll** [comment]
I've battled srutinizer down to 2 warnings about possible duplicated text (edit vs view). Hopefully that's sufficiently scrutinized. I'm used to forcing emacs to _avoid_ using tabs.

File: #helpers.php# [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		break;
+	default:
+		$output .= '<a href="' . $getter . '">' . $linktext . '</a>';
+		break;
+	}
+
+	return $output;
+
+}
+add_shortcode( 'wpghs', 'write_wpghs_link' );
```
Comment (mAAdhaTTah) [inline #helpers.php#]: What is this file?

File: helpers.php [hunk 2]
```diff
@@ -13,6 +13,7 @@
 function get_the_github_view_link() {
 	return '<a href="' . get_the_github_view_url() . '">' . apply_filters( 'wpghs_view_link_text', 'View this post on GitHub.' ) . '</a>';
 }
+add_shortcode( 'get_the_github_view_link', 'get_the_github_view_link' );
```
Comment (mAAdhaTTah) [inline helpers.php]: Let's kill off these shortcodes and just use the main one.

**Comment by mAAdhaTTah** [comment]
Sorry about the delay. Left two comments. Also fixed the test issue in `master`, so when this gets merged, we'll see if everything works.

**Comment by jonocarroll** [comment]
Updated. Removed the individual shortcodes. Still not passing Travis. Dropped the emacs temporary file (#helpers.php#) and excluded any others via .gitignore. Tested on my own site.

**Comment by mAAdhaTTah** [comment]
Yeah, travis breaking is being caused by something else. Thanks for your work on this!

**Pull request merged**